assets/src/components/rightPanel.tsx

```diff
@@ -1,35 +1,30 @@
-import React, { ReactElement } from "react"
-import { Ghost, Vehicle } from "../realtime.d"
+import React, { ReactElement, ReactNode } from "react"
 import NotificationDrawer from "./notificationDrawer"
-import PropertiesPanel from "./propertiesPanel"
 import SwingsView from "./swingsView"
 import { OpenView } from "../state/pagePanelState"
-import { usePanelStateFromStateDispatchContext } from "../hooks/usePanelState"
+import LateView from "./lateView"
 
-const RightPanel = ({
-  selectedVehicleOrGhost,
-}: {
-  selectedVehicleOrGhost?: Vehicle | Ghost | null
-}): ReactElement<HTMLElement> | null => {
-  const {
-    currentView: { openView },
-    closeView,
-  } = usePanelStateFromStateDispatchContext()
+type RightPanelProps = {
+  openView: OpenView
+  propertiesPanel?: ReactNode
+}
 
-  if (selectedVehicleOrGhost) {
-    return (
-      <PropertiesPanel
-        selectedVehicleOrGhost={selectedVehicleOrGhost}
-        onClosePanel={closeView}
-      />
-    )
-  } else if (openView === OpenView.Swings) {
+const RightPanel = ({
+  openView,
+  propertiesPanel,
+}: RightPanelProps): ReactElement | null => {
+  if (propertiesPanel !== undefined) {
+    return <>{propertiesPanel}</>
+  }
+  if (openView === OpenView.Swings) {
```

Comment:
**suggestion (non-blocking):** I think I sort of get why the first `if` statement is separated out rather than being part of the main series of `if` / `else` statements, but I personally feel like it would be more readable if it was all one series of `if` / `else`s.

Reply:
Okay I meant to not merge them yet, but I did, so I've fixed this in the latest force push.
13d46768878283a58025d4054f9e40d0f4197520